### ISSUE TITLE
`nvim_get_runtime_file()` returns duplicate entries for packages

### ISSUE THREAD

**Issue by nanotee** [neovim/neovim#14294]
- `nvim --version`:

    <details>
    <summary>Version</summary>

    ```
    NVIM v0.5.0-dev+1221-g9b2d4ff62
    Build type: RelWithDebInfo
    LuaJIT 2.1.0-beta3
    Compilation: /usr/bin/cc -O2 -g -Og -g -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wmissing-prototypes -Wimplicit-fallthrough -Wvla -fstack-protector-strong -fno-common -fdiagnostics-color=always -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -DMIN_LOG_LEVEL=3 -I/home/timothee/.local/bin/neovim-src/build/config -I/home/timothee/.local/bin/neovim-src/src -I/home/timothee/.local/bin/neovim-src/.deps/usr/include -I/usr/include -I/home/timothee/.local/bin/neovim-src/build/src/nvim/auto -I/home/timothee/.local/bin/neovim-src/build/include
    Compiled by timothee@timothee-laptop

    Features: +acl +iconv +tui
    See ":help feature-compile"

       system vimrc file: "$VIM/sysinit.vim"
      fall-back for $VIM: "/usr/local/share/nvim"

    Run :checkhealth for more info
    ```

    </details>

- `vim -u DEFAULTS` (version: ) behaves differently? N/A
- Operating system/version: Solus 4.2 Fortitude
- Terminal name/version: Konsole 20.12.3
- `$TERM`: xterm-256color

### Steps to reproduce using `nvim -u NORC`

```sh
nvim -u NORC
```

```vim
:call mkdir(stdpath('data') .. '/site/pack/test/start/test/testdir', 'p')
:quit
```

```sh
nvim -u NORC
```

```vim
:echo nvim_get_runtime_file('testdir/', v:true)
```

### Actual behaviour

Neovim returns a list with two `testdir/` entries

### Expected behaviour

Return only one entry like `globpath()` does:

```vim
:echo globpath(&rtp, 'testdir/', v:true, v:true)
```

**Comment by theHamsta** [comment]
@steelsojka

**Comment by shadmansaleh** [comment]
Fixed with #15632 @bfredl .
Should be closed ?
